jsettlers.mapcreator/src/jsettlers/mapcreator/control/UndoRedoHandler.java

```diff
@@ -0,0 +1,131 @@
+package jsettlers.mapcreator.control;
+
+import java.util.LinkedList;
+
+import jsettlers.mapcreator.data.MapData;
+import jsettlers.mapcreator.data.MapDataDelta;
+import jsettlers.mapcreator.main.window.EditorFrame;
+
+/**
+ * Handle Undo / Redo actions
+ * 
+ * @author Andreas Butti
+ *
+ */
+public class UndoRedoHandler {
+
+	/**
+	 * Max undo counts
+	 */
+	private static final int MAX_UNDO = 100;
```

Comment:
The reason for this is that undo deltas can get very big since they are not compressed. We will eventually run out of heap. We need to compress those deltas. 100 undo steps should be enough for all basic work.